### PR TITLE
Kodi binary add-ons: add visualization.matrix

### PR DESCRIPTION
**Pull request by sarbes**
Hi.

This PR adds my new Matrix viz (https://github.com/sarbes/visualization.matrix) to Kodi. I've tested in on a RPI 1, where it archived 40-50 fps.

This is a pre-release version to get some feedback in.

### CODE REVIEW

File: packages/mediacenter/kodi-binary-addons/visualization.matrix/package.mk [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2020-present Team LibreELEC (https://libreelec.tv)
+
+
```
Comment (chewitt) [inline packages/mediacenter/kodi-binary-addons/visualization.matrix/package.mk:4]: remove line (single space)

File: packages/mediacenter/kodi-binary-addons/visualization.matrix/package.mk [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+PKG_NAME="visualization.matrix"
+PKG_VERSION="v0.4.2"
+PKG_SHA256="2fb66404c43d6279a403d9ef9a2bc18d0944bae2033bc1cd7b1f78b5892d5c0c"
+PKG_REV="1"
+PKG_ARCH="any"
```
Comment (chewitt) [inline packages/mediacenter/kodi-binary-addons/visualization.matrix/package.mk:9]: remove

File: packages/mediacenter/kodi-binary-addons/visualization.matrix/package.mk [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+PKG_DEPENDS_TARGET="toolchain kodi-platform glm"
+PKG_SECTION=""
+PKG_SHORTDESC="visualization.matrix"
+PKG_LONGDESC="visualization.matrix"
+
+PKG_IS_ADDON="yes"
+PKG_ADDON_TYPE="xbmc.player.musicviz"
+
+if [ ! "$OPENGL" = "no" ]; then
+# for OpenGL (GLX) support
```
Comment (chewitt) [inline packages/mediacenter/kodi-binary-addons/visualization.matrix/package.mk:22]: indent

File: packages/mediacenter/kodi-binary-addons/visualization.matrix/package.mk [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  PKG_DEPENDS_TARGET="$PKG_DEPENDS_TARGET $OPENGL glew"
+fi
+
+if [ "$OPENGLES_SUPPORT" = yes ]; then
+# for OpenGL-ES support
```
Comment (chewitt) [inline packages/mediacenter/kodi-binary-addons/visualization.matrix/package.mk:27]: indent

**Comment by sarbes** [comment]
Will do. Except for the newline, it is a copy of the shadertoy package :).

I'll also update the URL, as it is now under the Team Kodi organization.

File: packages/mediacenter/kodi-binary-addons/visualization.matrix/package.mk [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (CvH) [inline packages/mediacenter/kodi-binary-addons/visualization.matrix/package.mk:1]:
```suggestion
# SPDX-License-Identifier: GPL-2.0
```
The LE code is under GPL2.0 so we obviously prefer to use it at new packages too.

**Comment by sarbes** [comment]
Done.

File: packages/mediacenter/kodi-binary-addons/visualization.matrix/package.mk [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+PKG_NAME="visualization.matrix"
+PKG_VERSION="v0.4.2"
+PKG_SHA256="2fb66404c43d6279a403d9ef9a2bc18d0944bae2033bc1cd7b1f78b5892d5c0c"
+PKG_REV="1"
+PKG_LICENSE="GPL"
+PKG_SITE="https://github.com/xbmc/visualization.matrix"
+PKG_URL="https://github.com/xbmc/visualization.matrix/archive/$PKG_VERSION.tar.gz"
+PKG_DEPENDS_TARGET="toolchain kodi-platform glm"
+PKG_SECTION=""
+PKG_SHORTDESC="visualization.matrix"
```
Comment (CvH) [inline packages/mediacenter/kodi-binary-addons/visualization.matrix/package.mk:13]:
```suggestion
```

sorry missed that, line could be removed

**Comment by CvH** [comment]
@sarbes tx for contribution

**Comment by CvH** [comment]
@sarbes we have builderrors at jenkins, any idea ?
```
1/5] Building CXX object CMakeFiles/visualization.matrix.dir/src/main.cpp.o
FAILED: CMakeFiles/visualization.matrix.dir/src/main.cpp.o
/home/jails/home/jenkins/LE/build2/workspace/Addons/All_Addons-RPi2/LibreELEC.tv/build.LibreELEC-RPi2.arm-9.80-devel/toolchain/bin/armv7ve-libreelec-linux-gnueabihf-g++  -DADDON_GLOBAL_VERSION_FILESYSTEM_USED -DADDON_GLOBAL_VERSION_GENERAL_USED -DADDON_GLOBAL_VERSION_GUI_USED -DADDON_GLOBAL_VERSION_MAIN_USED -DADDON_INSTANCE_VERSION_VISUALIZATION_USED -DBUILD_KODI_ADDON -DHAS_GLES=2 -Dvisualization_matrix_EXPORTS -I../lib -march=armv7ve -mtune=cortex-a7 -mabi=aapcs-linux -Wno-psabi -Wa,-mno-warn-deprecated -mfloat-abi=hard -mfpu=neon-vfpv4 -Wall -pipe  -O2 -fomit-frame-pointer -Os -DNDEBUG -fPIC   -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_HEAP=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_LINUX_MEMFD=1 -std=c++14 -MD -MT CMakeFiles/visualization.matrix.dir/src/main.cpp.o -MF CMakeFiles/visualization.matrix.dir/src/main.cpp.o.d -o CMakeFiles/visualization.matrix.dir/src/main.cpp.o -c ../src/main.cpp
../src/main.cpp:32:10: fatal error: kodi/libXBMC_addon.h: No such file or directory
   32 | #include "kodi/libXBMC_addon.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~
```
any idea ?

**Comment by HiassofT** [comment]
FYI: I'm getting the same error on a local RPi4 master build

**Pull request merged**